test/libsolidity/semanticTests/abiEncoderV2/dirty_memory_dynamic_array_and_calldata_string.sol

```diff
@@ -0,0 +1,71 @@
+pragma abicoder v2;
+
+contract C {
+    struct Data {
+        uint256[] a;
+        string b;
+    }
+
+    function makeDirty() internal
+    {
+        uint memStart = 0;
+        assembly { memStart := mload(0x40) }
+        uint memEnd = memStart + 0x800;
+
+        for (uint pos = memStart; pos < memEnd; pos += 0x20)
+            assembly { mstore(pos, not(0)) }
+    }
+
+
+    function testDirtyCalldata() public returns (bytes memory)
+    {
+        makeDirty();
+
+        Data memory data;
+        data.a = new uint256[](1);
+        data.a[0] = 0xFF;
+        data.b = "123456";
+        uint256[] memory array = new uint256[](2);
+        array[0] = 0xAF;
+        array[1] = 0xBF;
```

Comment:
Now that the memory and calldata cases are so similar, you could put this part in a helper function to make the test shorter.